app/scripts/analysis/tools/powercalc/PowercalcSidebarCtrl.ctrl.coffee

```diff
@@ -0,0 +1,304 @@
+					return
+
+				# extract data from data to population
+				index1 = data.header.indexOf(@chosenCols[0])
+				index2 = data.header.indexOf(@chosenCols[1])
+				@populations = {}
+				@populations[@chosenCols[0]] = []
+				@populations[@chosenCols[1]] = []
+				for row in data.data
+					@populations[@chosenCols[0]].push(row[index1])
+					@populations[@chosenCols[1]].push(row[index2])
+				console.log @populations
+
+
+			@msgService.broadcast 'powercalc:TwoTGUI_data',
+				populations:@populations
+				chosenCol:@chosenCols
+				chosenVar:@chosenVars
+				chosenlab:@chosenLabel
+		else if (@selectedAlgorithm is 'One-Sample (or Paired) t Test')
+			# if compare two different Variables, calculate sepaerately
+			if (@chosenLabel isnt "none") and (@chosenLabel isnt null)
+
+				#extract index if col
+				index = data.header.indexOf(@chosenCols)
+
+				#check if index if -1
+				if index is -1
+					console.log -1
+					return 
+
+				#extract data from container to population 
+				@populations = {}
+				@populations[@chosenVars] = []
+				for row in @container[@chosenVars]
+					@populations[@chosenVars].push(row[index])
+				console.log @populations
+
+			else
+				# extract data from data to population
+				index1 = data.header.indexOf(@chosenCols)
+				@populations = {}
+				@populations[@chosenCols] = []
+				for row in data.data
+					@populations[@chosenCols].push(row[index1])
+				console.log @populations
+
+
+			@msgService.broadcast 'powercalc:OneTGUI_data',
+				populations:@populations
+				chosenCol:@chosenCols
+				chosenVar:@chosenVars
+				chosenlab:@chosenLabel
+
+
+
+	updateAlgControls: () ->
+		#update algorithm method in local and broadcast to main control
+		#broadcast algorithms to main controller
+		@msgService.broadcast 'powercalc:updateAlgorithm',
+			@selectedAlgorithm
+
+
+	updateVar: (data) ->
+		index = data.header.indexOf(@chosenLabel)
+		@vars = []
+		@container = []
+		if index isnt -1
+			for row in data.data
+				if row[index] not of @container
+					@container[row[index]] = []
+
+				if row[index] not in @vars
+					@vars.push(row[index])
+
+				@container[row[index]].push(row)
+				#console.log (@container)
+
+		console.log @container
+		
+			
+
+	uniqueVals: (arr) -> arr.filter (x, i, a) -> i is a.indexOf x
+
+	parseData: (data) ->
+		@df = data
+		@dataService.inferDataTypes data, (resp) =>
+			if resp? and resp.dataFrame? and resp.dataFrame.data?
+
+				#update data types
+				for type, idx in @df.types
+					@df.types[idx] = resp.dataFrame.data[idx]
+
+				# update columns
+				@categoricalCols = []
+				@labelCol = ["none"]
+				id = 0
+				for header in @df.types
+					if header in ["number", "integer"]
+						@categoricalCols.push(@df.header[id])
+					else if header in ["string"]
+						@labelCol.push(@df.header[id])
+					id += 1
+			# @updateDataPoints(@df)
+
+			# @msgService.broadcast 'powercalc:updateDataPoints',
+			# 	dataPoints: @df
+
+	slidebar: () ->
```

Comment:
generally, the use of just a noun is a bad choice to name a function because function usually performs an action, so it's better practice to indicate in the function name which action it performs